beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/monitoring/BytesDeletedReporter.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.cleanup.monitoring;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.search.Search;
+
+import com.expediagroup.beekeeper.core.config.FileSystem;
+
+@Service
+public class BytesDeletedReporter {
+
+  public static final String METRIC_NAME = "bytes-deleted";
+  public static final String DRY_RUN_METRIC_NAME = "dry-run-" + METRIC_NAME;
+
+  private MeterRegistry meterRegistry;
+  private String metricName;
+
+  @Autowired
+  public BytesDeletedReporter(MeterRegistry meterRegistry,
+    @Value("${properties.dry-run-enabled}") boolean dryRunEnabled) {
+    this.meterRegistry = meterRegistry;
+    this.metricName = dryRunEnabled ? DRY_RUN_METRIC_NAME : METRIC_NAME;
+  }
+
+  public void report(long bytesDeleted, String fullyQualifiedTableName, FileSystem fileSystem) {
+    String fileSystemMetricName = String.join("-", fileSystem.toString()
+      .toLowerCase(), metricName);
+    Counter counter = Search.in(meterRegistry)
+      .name(fileSystemMetricName)
+      .tags(tags(fullyQualifiedTableName))
+      .counter();
+    if (counter == null) {
```

Comment:
Would this method ever be called by multiple threads? If so what would the effect be of registering the counter multiple times? My guess is that the last thread to exit would override the first one and we'd lose the first thread's metrics...

Reply:
Yes it could potentially be called by multiple threads. It looked like the `MeterRegistry` is thread safe.

Following the `register()` method through, we get to here: https://github.com/micrometer-metrics/micrometer/blob/f1fb973d6314cffac25059ee41926b487ba7bb25/micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java#L548

Reply:
So probably could even get away with not even searching for the counter, and just registering it each time?

Reply:
I don't know, wouldn't that register a new object on each report and thus you'd lose any previously stored values? (unless I'm misunderstanding you).

I think in our usage we do the deletes serially so I don't think we have multiple threads calling `report` so we're probably OK? Otherwise we'd need to make this synchronized or introduce locking.

Reply:
Indeed we do them serially. 

So in that link, the `getOrCreateMeter()` method uses synchronized to make sure that users can't create a new Meter while one is already being created. So if one thread finishes creating the same meter, then the next thread will just retrieve it. Does that make sense?

Reply:
And that ID used to identify a meter comprises the name, tags and meter type. So we're good on that front too as different each table will tag the meter differently.

Reply:
OK, yes, makes sense, so it sounds like we could call register each time or just leave the code like you have it.

Reply:
Haven't seen any examples of this no any documentation, so I think its probably fine to remove the check and just do the register.

Reply:
@abhimanyugupta07 here's the discussion about creating the counter